app/javascript/components/Payment/PaymentTypeSelection.js

```diff
@@ -1,39 +1,70 @@
 // @flow
 import React, { Component } from 'react';
 import { FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
 import PaymentMethodWrapper from '../ExpressDonation/PaymentMethodWrapper';
+import type { AppState } from '../../state';
+import type { PaymentType } from '../../state/fundraiser/types';
 
-export default class PaymentTypeSelection extends Component {
-  props: {
-    disabled?: boolean;
-    currentPaymentType?: string;
-    onChange: (paymentType: string) => void;
-    showDirectDebit: ?boolean;
-  };
+type Props = {
+  disabled?: boolean,
+  currentPaymentType?: string,
+  onChange: (paymentType: string) => void,
+  showDirectDebit: boolean,
+  directDebitOnly: boolean,
+};
+export class PaymentTypeSelection extends Component {
+  props: Props;
+
+  constructor(props: Props) {
+    super(props);
+  }
+
+  paymentTypes(): PaymentType[] {
+    const paymentTypes = [];
+
+    if (
+      !this.props.directDebitOnly ||
+      (this.props.directDebitOnly && !this.props.showDirectDebit)
```

Comment:
in the interests of readability, could this be wrapped in a method? `if ( noDirectDebit() )`